File: 백준/Silver/2606. 바이러스/바이러스.py
from collections import deque

N = int(input())
graph = [[] for _ in range(N + 1)]

E = int(input())
for _ in range(E):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (N + 1)
queue = deque([1])
visited[1] = True
count = 0

while queue:
    node = queue.popleft()
    for next in graph[node]:
        if not visited[next]:
            visited[next] = True
            queue.append(next)
            count += 1

print(count)

File: 백준/Silver/2606. 바이러스/바이러스.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const N = Number(input.shift());
const E = Number(input.shift());

const graph = Array.from({length: N + 1}, () => []);
for (let i = 0; i < E; i++) {
    const [a, b] = input[i].split(' ').map(Number);
    graph[a].push(b)
    graph[b].push(a)
}

const visited = new Array(N + 1).fill(false);
visited[1] = true

const queue = [1];
let count = 0;

while (queue.length > 0) {
    const node = queue.shift();
    for (let next of graph[node]) {
        if (!visited[next]) {
            visited[next] = true;
            queue.push(next);
            count++;
        }
    }
}

console.log(count);